src/middlewared/middlewared/plugins/apps/schema_normalization.py

```diff
@@ -103,9 +103,12 @@ async def normalize_gpu_configuration(self, attr_schema, value, complete_config,
             gpu['pci_slot']: gpu
             for gpu in await self.middleware.call('app.gpu_choices_internal') if not gpu['error']
         }
+
+        value['kfd_device_exists'] = self.kfd_exists
+
         if not any(gpu['vendor'] != 'NVIDIA' for gpu in gpu_choices.values()):
```

Comment:
I think we also want to change this line here, to read a bit better
if all(gpu['vendor'] == 'NVIDIA' for gpu in gpu_choices.values()):

Took me a minute to grasp my head around it.